src/main/java/net/snowflake/client/core/SFSessionInterface.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright (c) 2012-2021 Snowflake Computing Inc. All rights reserved.
+ */
+
+package net.snowflake.client.core;
+
+import java.sql.DriverPropertyInfo;
+import java.sql.SQLException;
+import java.util.List;
+import java.util.Properties;
+import net.snowflake.client.jdbc.SnowflakeSQLException;
+import net.snowflake.client.jdbc.SnowflakeType;
+import net.snowflake.client.jdbc.telemetry.Telemetry;
+
+/** Snowflake session implementation */
+public interface SFSessionInterface {
```

Comment:
nit: I know it is not your job but if you can complete all javadoc that will be great.

Reply:
Will do so once the interface is finalized in the PR.